pkg/diagnose/check.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package diagnose
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	forwarder "github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+)
+
+func init() {
+	diagnosis.Register("check-datadog", diagnose)
+}
+
+// Diagnose configured checks. Run them twice to facilitate rate-based metrics
+// which otherwise will fail. In the future perhaps it can be avoided for
+// non-rate checks. Currently diagnose is implemented to run in the CLI process,
+// in the next version will connect to the running agent service to get diagnoses
+// for scheduled checks.
```

Comment:
Do we need to run the check twice for diagnose ?

Reply:
Old comment. I need to fix it. It was original implementation and @alopezz already mention that and I have addressed. Will fix the comment. Thank you.